src/Cipher.ts

```diff
@@ -257,35 +257,19 @@ class Decipher extends CipherCommon {
 export function createDecipher(
   algorithm: string,
   password: BinaryLike,
-  options?: Stream.TransformOptions
+  options?: Record<string, unknown> | Stream.TransformOptions
```

Comment:
Why the untyped return type (Record<string, unknown>)?

Is there no way we can also type that?

Reply:
If you expand the diff there are overloads commented up above. Without those overloads exposed as well as intermediate typescript guard functions that execute at runtime there is no way of getting accurate typing here. My change makes the types more accurate without doing that by allowing users to use any object, because `Stream.TransformOptions` is only one of 4 overloads and not the accurate type to use here.

Reply:
hm I see, but can't we create a joined type of the possible overloads instead of making this even more lose?

Reply:
If we create a joined type we will need a type guard to use any one of the joins. Those enhancements are unrelated to the PR since they reflect project wide decisions and tradeoffs in how to handle type safety (use runtime guards vs. prefer looseness) made before my contributions. One way we can avoid runtime guards is by chaining `Type A | Type B | Type C | AnyObject`.

Reply:
I was wrong here and able to fix in #247